TicketsDemo/App_Start/NinjectWebCommon.cs

```diff
@@ -65,17 +66,28 @@ private static IKernel CreateKernel()
         private static void RegisterServices(IKernel kernel)
         {
             kernel.Bind<ITicketRepository>().To<TicketRepository>();
-            kernel.Bind<ITrainRepository>().To<TrainRepository>();
+            kernel.Bind<ITrainRepository>().To<MongoTrainRepository>();
+            kernel.Bind<IRepresentativeRepository>().To<RepresentativeRepository>(); ;
 
             kernel.Bind<IRunRepository>().To<RunRepository>();
             kernel.Bind<IReservationRepository>().To<ReservationRepository>();
 
             kernel.Bind<ISchedule>().To<Schedule>();
-            kernel.Bind<ITicketService>().To<TicketService>();
+            //kernel.Bind<ITicketService>().To<TicketService>();
+
+            kernel.Bind<ITicketService>().To<TicketServiceLoggingDecorator>();
+            kernel.Bind<ITicketService>().To<TicketService>().WhenInjectedExactlyInto<TicketServiceLoggingDecorator>();
             kernel.Bind<IReservationService>().To<ReservationService>();
 
+            kernel.Bind<IPriceCalculationStrategy>().ToMethod<FinalPriceCalculationStrategy>(ctx => {
```

Comment:
equivalent to

kernel.Bind<IPriceCalculationStrategy>().To<FinalPriceCalculationStrategy>();

kernel.Bind<IPriceCalculationStrategy>().To<DefaultPriceCalculationStrategy>().WhenInjectedExactlyInto<FinalPriceCalculationStrategy>();
kernel.Bind<IPriceCalculationStrategy>().To<BookingPriceCalculationStrategy>().WhenInjectedExactlyInto<FinalPriceCalculationStrategy>();